### PR TITLE
ENH: Add min values comparison for floating point

### PR DESCRIPTION
**Pull request by howjmay**
Add the similar implementation for minimum value comparison.

### CODE REVIEW

File: numpy/core/src/common/simd/avx2/math.h [hunk 1]
```diff
@@ -56,4 +56,23 @@ NPY_FINLINE npyv_f64 npyv_maxp_f64(npyv_f64 a, npyv_f64 b)
     return _mm256_blendv_pd(a, max, nn);
 }
 
+// Minimun, natively mapping with no guarantees to handle NaN.
```
Comment (seiko2plus) [inline numpy/core/src/common/simd/avx2/math.h:59]:
```suggestion
// Minimum, natively mapping with no guarantees to handle NaN.
```
Just a typo needs to get fixed across all the C comments in this patch.

**Review by seiko2plus — APPROVED** [review]
LGTM, Thank you!

File: numpy/core/src/common/simd/avx2/math.h (hunk 1 above)
Comment (HowJMay) [reply numpy/core/src/common/simd/avx2/math.h]: fixed!

**Comment by seiko2plus** [comment]
close/open to re-run travis-ci.

**Comment by mattip** [comment]
This seems to be a follow-on PR to adding max value comparison. Which PR was that?

**Comment by mattip** [comment]
ahh, #18863

File: numpy/core/src/_simd/_simd.dispatch.c.src [hunk 2]
```diff
@@ -390,6 +390,8 @@ SIMD_IMPL_INTRIN_1(@intrin@_@sfx@, v@sfx@, v@sfx@)
 #if @fp_only@
 SIMD_IMPL_INTRIN_2(max_@sfx@, v@sfx@, v@sfx@, v@sfx@)
 SIMD_IMPL_INTRIN_2(maxp_@sfx@, v@sfx@, v@sfx@, v@sfx@)
+SIMD_IMPL_INTRIN_2(min_@sfx@, v@sfx@, v@sfx@, v@sfx@)
+SIMD_IMPL_INTRIN_2(minp_@sfx@, v@sfx@, v@sfx@, v@sfx@)
```
Comment (mattip) [inline numpy/core/src/_simd/_simd.dispatch.c.src:394]: style nit: just above there is a repeat clause for 4 values, here all 4 are spelled out explicitly. Can we unify the approach?

**Comment by mattip** [comment]
I understand correctly that these two PRs only add the intrinsic, there will be a follow-on one to use it?

File: numpy/core/src/_simd/_simd.dispatch.c.src (hunk 2 above)
Comment (seiko2plus) [reply numpy/core/src/_simd/_simd.dispatch.c.src:394]: maybe a repeater similar to
https://github.com/numpy/numpy/blob/9a889942f7e83d1b7fc474094061c0ab9e41d26b/numpy/core/src/_simd/_simd.dispatch.c.src#L379-L388

**Comment by seiko2plus** [comment]
`npyv_max_[f32, f64]`, `npyv_min_[f32, f64]` are usually going to be used within custom kernels since both intrinsics don't guarantees handle NaN. while `npyv_minp_[f32, f64]` and `npyv_maxp_[f32, f64]` can be used directly for optimizing [numpy.fmax](https://numpy.org/doc/stable/reference/generated/numpy.fmax.html).

File: numpy/core/src/_simd/_simd.dispatch.c.src (hunk 2 above)
Comment (HowJMay) [reply numpy/core/src/_simd/_simd.dispatch.c.src:394]: So let me refactor to this way

Comment (HowJMay) [reply numpy/core/src/_simd/_simd.dispatch.c.src]: done

**Comment by mattip** [comment]
Thanks @HowJMay

**Comment by HowJMay** [comment]
Thank you @mattip and @seiko2plus 
Would you guys mind me adding `max`, `min` for signed and unsigned integers, and `floor`, `ceil`and `negate` operations to simd sets?

**Comment by seiko2plus** [comment]
@HowJMay,
yes, that would great. feel free to add any set of intrinsic as long as it has an equivalent ufunc/umath function.

File: numpy/core/tests/test_simd.py [hunk 3]
```diff
@@ -317,11 +317,6 @@ def test_square(self):
         assert square == data_square
         
     def test_max(self):
-        """
-        Test intrinics:
-            npyv_max_##SFX
-            npyv_maxp_##SFX
-        """
```
Comment (seiko2plus) [inline numpy/core/tests/test_simd.py:324]: I didn't see these changes during my latest review.

File: numpy/core/tests/test_simd.py [hunk 4]
```diff
@@ -346,6 +341,31 @@ def test_max(self):
             _max = self.max(vdata_a, vdata_b)
             assert _max == data_max
 
+    def test_min(self):
```
Comment (seiko2plus) [inline numpy/core/tests/test_simd.py:344]: the following are missing too
```python
"""
Test intrinics:
    npyv_min_##SFX
    npyv_minp_##SFX
"""
````
I think these changes made during your latest push

File: numpy/core/tests/test_simd.py (hunk 3 above)
Comment (HowJMay) [reply numpy/core/tests/test_simd.py:324]: Sorry for removing this accidentally
I will fix this in the following PR of maximum and minimum

**Pull request merged**